astarte/device/device.py

```diff
@@ -92,22 +95,22 @@ def __init__(self,
             it is not recommended to leave this `true` in production.
             Defaults to `false`, if `true` the device will ignore SSL errors during connection.
         """
-        self.__device_id = device_id
-        self.__realm = realm
-        self.__pairing_base_url = pairing_base_url
-        self.__persistency_dir = persistency_dir
-        self.__credentials_secret = credentials_secret
-        self.__jwt_token = None
-        self.__is_crypto_setup = False
-        self.__introspection = Introspection()
-        self.__is_connected = False
-        self.__loop = loop
-        self.__ignore_ssl_errors = ignore_ssl_errors
-
-        self.on_connected = None
-        self.on_disconnected = None
+        self.__device_id: str = device_id
+        self.__realm: str = realm
+        self.__pairing_base_url: str = pairing_base_url
+        self.__persistency_dir: str = persistency_dir
+        self.__credentials_secret: str = credentials_secret
+        self.__jwt_token: Optional[str] = None
+        self.__is_crypto_setup: bool = False
```

Comment:
I see that in some places the type hinting is used, in others it's not.
Is this PR proposing to use it only where needed, or would it place it everywhere?
On some of these, like the `bool` the type is already clear. Using type hinting there might just make uncecessary clutter.